### PR TITLE
Adds new barebones spring security example. Adds filter to set the authentication with Spring Security when there's already an Account present.

### LINKED ISSUE #450: Redirect loop with basic Spring Security settings
If you have an app using `stormpath-default-spring-boot-starter` and no controllers with the most basic Spring Security configuration: `http.apply(StormpathWebSecurityConfigurer.stormpath())`, you can get stuck in a redirect loop.

Steps to reproduce:

1. Launch the basic app described above (connected to a valid Stormpath Application)
2. Launch Google Chrome (or Firefox or Safari)
3. Browse to: `http://localhost:8080` (you will be redirected to `/login` as expected)
4. Login
5. Quit the browser
6. Re-launch the browser
7. Browse to: `http://localhost:8080`
8. You will get the redirect loop error

Theory is that the `account` cookie is still present which identifies you to Stormpath as being logged in. However Spring Security does NOT have you as logged in. So, browsing to `/` redirects to `/login` because of Spring Security. Stormpath thinks your logged in and redirects back to `/`. 

StackO reference here: http://stackoverflow.com/questions/34886331/stormpath-spring-boot-redirect-loop/34888978#34888978

### PR DESCRIPTION
**Pull request by dogeared**
resolves #450

### CODE REVIEW

**Comment by lhazlewood** [comment]
LGTM

File: examples/spring-boot-webmvc-spring-security/pom.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                <artifactId>maven-deploy-plugin</artifactId>
+                <version>2.7</version>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (mrioan) [inline examples/spring-boot-webmvc-spring-security/pom.xml]: We already have a Spring Boot Spring Security WebMVC: examples/spring-security-spring-boot-webmvc. I do not think we need a new one

Comment (lhazlewood) [inline examples/spring-boot-webmvc-spring-security/pom.xml]: Ah good point.  I forgot about this.

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 2]
```diff
@@ -186,6 +194,9 @@ public void init(HttpSecurity http) throws Exception {
                     .passwordParameter("password")
                     .and().authorizeRequests()
                     .antMatchers(loginUriMatch).permitAll();
+
+                http
+                    .addFilterBefore(springSecurityResolvedAccountFilter, AnonymousAuthenticationFilter.class);
             }
```
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: You are basically trying to get `http.addFilterBefore(springSecurityResolvedAccountFilter, AnonymousAuthenticationFilter.class);` executed only if `loginEnabled` is true. Can you please add a comment saying something like "We need to add the springSecurityResolvedAccountFilter whenever we have our login enabled in order to fix https://github.com/stormpath/stormpath-sdk-java/issues/450"

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/SpringSecurityResolvedAccountFilter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.Assert;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class SpringSecurityResolvedAccountFilter extends HttpFilter implements InitializingBean {
```
Comment (mrioan) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/filter/SpringSecurityResolvedAccountFilter.java]: Please add this comment

/**
* Fix for https://github.com/stormpath/stormpath-sdk-java/issues/450
*
* @version 1.0.RC.8.1 and a commene
**/

**Comment by mrioan** [comment]
@dogeared a few comments. PR not approved yet.

**Comment by dogeared** [comment]
I renamed the new example. I think it's useful as it's a barebones spring security example. Everything is locked down. There's no controllers. Aside from being useful to the community, having such an example would've caught the redirect loop issue. Let me know if you still want me to remove it.

**Comment by mrioan** [comment]
@dogeared The new example is still a webmvc example. I think it is misleading not to include that in the name

**Pull request merged**